### ISSUE TITLE
npm list command failed

### ISSUE THREAD

**Issue by cjsaylor** [jadjoubran/webdash-package-json#6]
I get the following error from the console (and the plugin never completes load).

```bash
Webdash running on port http://localhost:3456!
exec error: Error: Command failed: npm list --depth=0 --json --silent
```

The command itself works when I execute from the command line (on the same path as the `serve` command).

Additionally, the node server never responds when this error occurs, so the request stays in pending status.

Node: v8.10.0 (carbon)
npm: 5.6.0

**Comment by jadjoubran** [comment]
Thanks for reporting!
When you run it from the command line, do you get an `npm update available` message at the end?
<img width="396" alt="screen shot 2018-03-26 at 4 57 32 pm" src="https://user-images.githubusercontent.com/2265232/37910712-da1774f4-3116-11e8-9bd8-3d20484059df.png">

**Comment by cjsaylor** [comment]
No warning after running the command.

**Comment by jadjoubran** [comment]
That's really weird
It's failing [here on Line 40](https://github.com/jadjoubran/webdash-package-json/blob/master/api.js#L40)
But the error message is not that useful

Can you please debug by following these steps:
1. Open `node_modules/webdash-package-json/api.js` in your text editor
2. Remove `--silent` from Line 11 for `commands.list`
3. Restart the `webdash serve` and check the console if you've got a more detailed error message

Thanks!

**Comment by cjsaylor** [comment]
Serve output:

```
Webdash running on port http://localhost:3456!
exec error: Error: Command failed: npm list --depth=0 --json
npm ERR! peer dep missing: popper.js@^1.12.9, required by bootstrap@4.0.0
```

So, what it looks like to me is the command is giving a non-zero exit code even with `--silent` option:

```bash
npm list --depth=0 --json --silent
echo $?
1
```

EDIT:

I'm intentionally not satisfying that peer dependency because I don't use any of the things `popper` is used from out of bootstrap.

**Comment by jadjoubran** [comment]
indeed! Because of the missing peer dependency
Can you try running it with `--no-interactive` and see if it exits with zero

**Comment by cjsaylor** [comment]
Still get a non-zero exit code with `--no-interactive`.

**Comment by jadjoubran** [comment]
I had the same error with the `outdated` command here: https://github.com/jadjoubran/webdash-package-json/blob/master/api.js#L82
If there are updates, npm would return a non-zero status code
I'll apply the same fix for this one
Thanks for catching it!
I'll fix it tomorrow, tag a release and update this issue so you can give it a try

**Comment by jadjoubran** [comment]
Alright can you please test with this preminor version:
`npm install webdash-package-json@v1.1.0-0 --save-dev`

**Comment by cjsaylor** [comment]
```
Webdash running on port http://localhost:3456!
There
TypeError: Cannot read property 'trees' of undefined
    at exec (/Users/christophersaylor/go/src/github.com/cjsaylor/boxmeup-ui/node_modules/webdash-package-json/api.js:47:46)
    at ChildProcess.exithandler (child_process.js:282:5)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Socket.stream.socket.on (internal/child_process.js:346:11)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at Pipe._handle.close [as _onclose] (net.js:567:12)
```

**Comment by jadjoubran** [comment]
This should work:
`npm install webdash-package-json@v1.1.0-2 --save-dev`

I just tried it locally after installing bootstrap without satisfying its peer dependency

**Comment by cjsaylor** [comment]
I don't get an error anymore, but the response has no results:

```
GET /api/package-json/package-json HTTP/1.1
Host: localhost:3456
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.186 Safari/537.36
Accept: */*
Referer: http://localhost:3456/
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9

HTTP/1.1 200 OK
X-Powered-By: Express
Access-Control-Allow-Origin: *
Content-Type: application/json; charset=utf-8
Content-Length: 13
ETag: W/"d-MoNXnKuD0ORPMCkqEFx5iouAPNo"
Date: Tue, 27 Mar 2018 00:00:18 GMT
Connection: keep-alive

{"result":{}}
```

**Comment by jadjoubran** [comment]
<img width="901" alt="screen shot 2018-03-27 at 12 42 59 pm" src="https://user-images.githubusercontent.com/2265232/37959706-6695216a-31bc-11e8-899b-a77fa2963fa9.png">

I tried reproducing your issue, and I see the errors here
but when I pass in `--silent` they're gone
so now I'm thinking maybe it has to do with `npm`? I'm running `npm 5.7.1` would you mind updating from `5.6.0`?

**Comment by cjsaylor** [comment]
I updated to latest npm and the issue persists.

The code  if you want to try is open source: https://github.com/cjsaylor/boxmeup-ui

webdash.json:

```json
{
    "src": "./src/",
    "dist": "./build/",
    "jsBudgetLimit": 200,
    "jsBudgetPath": "./build/**/*.js",
    "manifestPath": "./build/manifest.json",
    "readableAssets": {
        "./public/": "/"
    }
}
```

**Comment by jadjoubran** [comment]
Awesome, I was able to reproduce it and fix it
<img width="510" alt="screen shot 2018-03-27 at 2 53 00 pm" src="https://user-images.githubusercontent.com/2265232/37965683-9053b126-31ce-11e8-9f8b-e3220bed078a.png">

This created another issue which is unmet peer deps now show up without any version
I'm working on excluding them

**Comment by jadjoubran** [comment]
I can confirm it's working with `v1.1.0-3`

`npm install webdash-package-json@v1.1.0-3 --save-dev`

**Comment by cjsaylor** [comment]
Confirmed it is working! Thanks!

**Comment by jadjoubran** [comment]
Awesome 🎉 
This will be available in `v1.2` tonight
